### PR TITLE
Make diagonal movement faster (1.41x as long as cardinal, rather than 2x)

### PR DESCRIPTION
**Pull request by moxian**
## What Does This PR Do
Speeds up diagonal movement from its current state of being 2x slower than cardinal to a more natural 1.41x (that is, square root of 2)
This is a real gameplay change, not a visual-only.

## Why It's Good For The Game
A recent #17872 changed the visuals (but not the mechanics) for diagonal gliding to be smooth instead of the old jump+wait. And even though there was no gameplay impact in that, it now *feels* slower.
This is very similar to the feedback of the gliding being originally introduced - see #13381. Moving for 1 second+waiting for an extra second feels faster than moving for 2 seconds.

However, now that we have an accurate view of the real move speed, *and* that we realize it is indeed slower than you'd expect to have "in real life", it would make sense to adjust it to be more "realistic".
(something something math, something something square root of two)

There's currently no reason to ever use diagonals, which is a shame. This pr does give such reason.

## Images of changes
20220528-010723-dreamseeker

## Changelog
:cl:
tweak: Diagonal moves now takes 1.41x as long as cardinal ones (rather than 2x)
/:cl:

### CODE REVIEW

**Comment by Nerfection** [comment]
The discrepancy between regular speed horizontal and vertical movement makes it quite jarring now, as visually your character now has two distinct movement speeds. I massively preferred the previous move and pause system, though I like that diagonal movement is a bit less slow now.

**Comment by Lazyman310** [comment]
Moving diagonally now feels slower than it used to be, and is quite jarring even if it is technically faster. I thought that my game was lagging until I realized it was this

**Comment by TrashDoxx** [comment]
excuse me :trollface:
image

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -183,13 +183,14 @@
 	if(!(direct & (direct - 1))) // cardinal direction
 		. = mob.SelfMove(n, direct, delay)
 	else // diagonal movements take twice as long
-		. = mob.SelfMove(n, direct, delay * 2)
+		var/diag_delay = delay * 1.412
```
Comment (hal9000PR) [inline code/modules/mob/mob_movement.dm:186]:
```suggestion
		var/diag_delay = delay * SQRT_2
```

**Review by hal9000PR — APPROVED** [review]

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**